### ISSUE TITLE
Error 201

### ISSUE THREAD

**Issue by Kingsy** [nwestfal/pylotro#8]
So I am having an issue getting the launcher to work at all. Here are the steps I have taken.

1. Download lotrolive.exe from lotro.com
2. Install lotro using wine-3.0 on ubuntu 18.04
3. git clone this repo and install via python setup.py install
4. Open pylotro -> tools -> patch and run the patch on the game, which completed successfully.
5. Click login.

the error is

"Can't open the data files. Check that they exist and that you have permission to write to them. The program will now exit. [201]"

After pressing ok this error appears again and then it exits.

The message left in the pylotro window is

Game Client: lotroclient.exe
0009:err:ole:CoUninitialize Mismatched CoUninitialize

*** Finished ***

Is there something I am doing wrong?

Thanks

**Comment by nwestfal** [comment]
Hi,

I'm pretty sure the problem is that you need to be using lotroclient_awesomium.exe instead of lotroclient.exe.  See the 0.2.5 notes in the ChangeLog.  Since lotroclient_awesomium.exe seems to automatically get downloaded when patching, it may be enough to just cd to your game directory, and run:

mv TurbineLauncher.exe.config TurbineLauncher.exe.config.orig
wget http://installer.lotro.turbine.com/lotro/lotro_client/TurbineLauncher.exe.config

BTW that error you see in the pylotro window occurs for lotroclient_awesomium.exe as well and I see it after I exit the game.  Seems harmless and is probably is not related.

Neal

**Comment by Kingsy** [comment]
Thanks for the reply!
Yeah I have tried lotroclient.exe and lotroclient_awesomium.exe and both seem to give that error so I guess thats good, like you said probably not related.

However I did try to backup the TurbineLauncher.exe.config like you mentioned then the wget and still get the same error

"Can't open the data files. Check that they exist and that you have permission to write to them. The program will now exit. [201]"

The output in the launcher is still the same but as you mentioned its probably not related.

Game Client: lotroclient_awesomium.exe
0009:err:ole:CoUninitialize Mismatched CoUninitialize

Not really sure what else I can try.

**Comment by nwestfal** [comment]
This page mentions some dependencies that might be needed:

https://www.lotro.com/forums/showthread.php?654273-Windows-XP-and-Vista-Launcher-Issues-Solution

Try downloading the awesomium_dependencies.zip

There is also this page that looks to be rather recent, but not sure there is much there to be gleaned for Linux:

https://help.standingstonegames.com/hc/en-us/articles/115002137708-Game-error-201-Can-t-open-the-data-files-Check-that-they-exist-and-that-you-have-permission-to-write-to-them-The-program-will-now-exit-201-

Only other suggestion I can think of would be to try using the TurbineLauncher.exe and let it run through it's patch routine.  Some people have even reported that just using TurbineLauncher.exe runs fine in wine but it's always been hit and miss for me, like maybe working 1 in 5 times, but usually crashes.

**Comment by Kingsy** [comment]
Yeah i have tried those new files and the dependencies still the same error.
I have had no luck using the TurbineLauncher through wine either, crashes everytime as far as I can tell but I will keep trying.
I hear on a few posts that potentially some files are not pulled in by the pylotro patch routine which can cause this, for example I hear when it talks about being unable to access game files its talking about .dat files however if I look in the game dir I only have two of these.

ls -la *.dat*
-rw-rw-r-- 1 chris chris 10130464 Jun  6 22:34 icudtl.dat
-rw-rw-r-- 1 chris chris    79174 Jun  6 22:33 unins000.dat

Some people mention ,datx files too which obviously I don't have, do you have those on your working installation?
Some have said you need a seperate windows PC, let the game install through the native launcher then 
manually copy the .dat / .datx files over manually, any truth in this?

**Comment by nwestfal** [comment]
Here's a list of the .dat files in my game directory:

-rwxrwxrwx 1 neal users  650117120 Jun  6 15:18 client_anim.dat
-rwxrwxrwx 1 neal users  344981504 Jun  6 15:18 client_cell_1.dat
-rwxrwxrwx 1 neal users  331409848 Jun  6 15:18 client_cell_2.dat
-rwxrwxrwx 1 neal users  477245020 Jun  6 15:18 client_cell_3.dat
-rw-r--r-- 1 neal users  409995676 Jun  6 15:18 client_cell_4.dat
-rwxrwxrwx 1 neal users  701322028 Jun  6 15:18 client_gamelogic.dat
-rwxrwxrwx 1 neal users  480247808 Jun  6 15:18 client_general.dat
-rwxrwxrwx 1 neal users 4287627264 Jun  6 15:18 client_highres.dat
-rwxrwxrwx 1 neal users 1087658200 Jun  6 15:18 client_local_English.dat
-rwxrwxrwx 1 neal users   23068672 Jun  6 15:18 client_map_1.dat
-rwxrwxrwx 1 neal users   23068672 Jun  6 15:18 client_map_2.dat
-rwxrwxrwx 1 neal users   23068672 Jun  6 15:18 client_map_3.dat
-rw-r--r-- 1 neal users   23068672 Jun  6 15:18 client_map_4.dat
-rwxrwxrwx 1 neal users 2648702976 Jun  6 15:18 client_mesh.dat
-rwxrwxrwx 1 neal users 4287627264 Jun  6 15:18 client_sound.dat
-rwxrwxrwx 1 neal users 4287627264 Jun  6 15:18 client_surface.dat
-rw-r--r-- 1 neal users   10130464 May 31  2017 icudtl.dat
-rwxrwxrwx 1 neal users     112355 Apr 23  2013 unins000.dat

And .datx files:

-rwxrwxrwx 1 neal users 3855613952 Jun  6 15:18 client_highres_aux_1.datx
-rw-r--r-- 1 neal users   84934656 Jun  6 15:18 client_sound_aux_1.datx
-rwxrwxrwx 1 neal users 2222981120 Jun  6 15:18 client_surface_aux_1.datx

So yeah looks like you are missing a bunch.  What's interesting is most of these are dated June 6th and I just ran a patch yesterday from pylotro so it seems to pull them down for me.  I haven't really looked closely at the patch code, but maybe it only tries to download newer versions of files that are already present.  Maybe you could try creating dummy files of all those using the touch command, then try the patch routine again.

Like cd to the game directory, and run:

touch client_anim.dat
touch client_cell_1.dat
etc.

**Comment by Kingsy** [comment]
Yeah just a touch with the filename and another patch run didnt work.
I'll try and install on a windows pc and transfer the dats over. Hopefully that works.. 
Man, certainly have to go round the houses to make this work. I'll keep this thread updated if that works, hopefully it will help someone else

**Comment by nwestfal** [comment]
For what it's worth, I just installed a fresh copy using lotrolive.exe, and found that the game folder only contained 1.9 GB while my old installation contains 27 GB of data.  After several attempts to launch the TurbineLauncher, it downloaded Akamai NetSession and is now downloading the client files.  There seems to be no rhyme or reason to when the TurbineLauncher works or crashes, but at the moment it seems to be downloading.  Once it's done I'll post here whether or not it works.

**Comment by Kingsy** [comment]
What version of wine are you running?
I have had no luck getting the launcher to work on Linux, it just doesn't establish the connection with Akamai.

**Comment by nwestfal** [comment]
3.4 on OpenSUSE 42.2.  It's been awhile since I updated.  But the launcher was mostly crashing on me too.  It's just that every once in a while out of the blue it works.

**Comment by nwestfal** [comment]
It did finish downloading all the files, and the game directory had the full 27 GB when it completed.  The game loaded fine with pylotro after that.

**Comment by Kingsy** [comment]
Just going to try both methods today see which or hopefully both work. I'll basically just try restarting the launcher until it springs into life.
Curious, once you install the files via the TurbineLauncher I am assuming you still need to do a patch run with pylotro before launch?

**Comment by Kingsy** [comment]
Sorry for the delay but hopefully this will help anyone else with the same issues.
I can confirm that I managed to get both methods working.
The turbineLauncher does indeed work, you just need to try it multiple times from a fresh installation. From what I saw it crashes as each stage of the launcher inits. And lastly when it attempts to establish a connection with the servers. If you crash out and reload from this point it worked for me.
If all else fails installing on a windows PC and copying the relevant game files also does work.

nwestfal thankyou for your help
